### ISSUE TITLE
security/acme-client: Fix wrong permissions in 'Update to local Unifi keystore' automation

### ISSUE THREAD

**Issue by mdbraber** [opnsense/plugins#4417]
**Important notices**
Before you add a new report, we ask you kindly to acknowledge the following:

- [x] I have read the contributing guide lines at https://github.com/opnsense/plugins/blob/master/CONTRIBUTING.md
- [x] I have searched the existing issues, open and closed, and I'm convinced that mine is new.
- [x] The title contains the plugin to which this issue belongs

**Describe the bug**
When deploy an certificate obtained via acme-client, the permissions on the key store are wrong. Also see here: https://forum.opnsense.org/index.php?topic=43556.msg216736#msg216736

**To Reproduce**
(using the Unifi plugin from the mimugmail repo)
1. Obtain a certificate
2. Use the "Deploy to Unifi" command after obtaining the certificate
3. Try to access Unifi

**Expected behavior**
Expected to have a clean reload

**Suggested fix**

What could help is to change this line here: https://github.com/opnsense/plugins/blob/80c2623bd581f4586b09eb54ae30b2e0965cf60c/security/acme-client/src/opnsense/mvc/app/library/OPNsense/AcmeClient/LeAutomation/AcmeUnifi.php#L42 to

`$this->acme_env['DEPLOY_UNIFI_RELOAD'] = 'service unifi restart';` 

change into

`$this->acme_env['DEPLOY_UNIFI_RELOAD'] = 'chown unifi:wheel ' + (string)$this->config->acme_unifi_keystore + '; service unifi restart'`

I've considered having a separate `chown` command, but order would not be expected to be the same and that solution seems brittle. Post-hooks in acme.sh are run before deploy hooks in acme.sh so that won't work either. Also changing the deploy hook in acme.sh is very specific to OPNsense and will possibly not be accepted in the distribution (which would be fair). So I think the above would be the best place to fix this.

**Comment by adn77** [comment]
Thank you for getting to the bottom of it. I noticed some weirdness with the certificate deployment as well in recent versions but didn't think much of it. 

Either acme.sh or Unifi changed the user they are running under. 

I proposed an addition to the upstream project. If that finds support, I am going to fix it there. 
Maybe we'll get around changing anything here then.

**Comment by adn77** [comment]
I created an acme.sh PR here
https://github.com/acmesh-official/acme.sh/pull/6168

You can simply apply the changes in `/usr/local/share/examples/acme.sh/deploy/unifi.sh` until it gets merged.

**Comment by adn77** [comment]
The PR has been merged to the acme.sh Dev branch. Should be in the next release of acme.sh 😄

**Comment by fraenki** [comment]
Thanks for fixing the upstream issue. This will be available in OPNsense when a new version of acme.sh was released.

I'll close this issue now, as manually applying the acme.sh PR solves this issue for the time being.

**Comment by asche77** [comment]
unfortunately, the upstream fix in acme.sh v3.1.1 has not fixed the issue for me (OPNsense 25.1.7_4).

Running the ACME automation sets the user at root:wheel:

```
% ls -la /usr/local/share/java/unifi/data/keystore
-rw-r-----  1 root wheel 5974 May 25 21:33 /usr/local/share/java/unifi/data/keystore

```

Doing a manual chown (and restart of unifi) works:

```
chown unifi:wheel /usr/local/share/java/unifi/data/keystore
chown unifi:wheel /usr/local/share/java/unifi/data/keystore_original
```

**Comment by adn77** [comment]
For some reason the PR was edited before merged. v3.1.1 contains code which doesn't work on *BSD.
The PR linked above tries to remedy the changes.

**Comment by adn77** [comment]
this is now in ACME.sh:master
https://github.com/acmesh-official/acme.sh/pull/6604

 ... almost there 😃

**Comment by adn77** [comment]
Now finally released as v3.1.2 which is included in opnsense v25.7.5

**Comment by asche77** [comment]
Thank you!
